serde_derive/src/internals/case.rs

```diff
@@ -6,6 +6,9 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+//! Utility code to assist converting the case of the source (e.g. `my-field`, `MY_FIELD`) to the
```

Comment:
What distinction is this making between utility code and ordinary code? I would recommend avoiding the word utility in naming or documenting anything. [https://blog.golang.org/package-names#TOC_5.](https://blog.golang.org/package-names#TOC_5.) is relevant.